docs/Account.md

```diff
@@ -0,0 +1,331 @@
+
+This message is consumed by the `execute` method, which acts as a single entrypoint for all user interaction with any contract, including managing the account contract itself. That's why if you want to change the public key controlling the Account, you would send a transaction targeting the very Account contract:
+
+```python
+await signer.send_transaction(account, account.contract_address, 'set_public_key', [NEW_KEY])
+```
+
+Notice that Signer's `send_transaction` calls `execute` under the hood.
+
+Or if you want to update the Account's L1 address on the `AccountRegistry` contract, you would 
+
+```python
+await signer.send_transaction(account, registry.contract_address, 'set_L1_address', [NEW_ADDRESS])
+```
+
+You can read more about how messages are structured and hashed in the [Account message scheme  discussion](https://github.com/OpenZeppelin/cairo-contracts/discussions/24).
+
+
+## API Specification
+
+This in a nutshell is the Account contract public API:
+
+```cairo
+func get_public_key() -> (res: felt)
+func get_address() -> (res: felt)
+func get_nonce() -> (res: felt)
+
+func set_public_key(new_public_key: felt)
+func initialize(_address: felt)
+
+func is_valid_signature(hash: felt,
+        signature_len: felt,
+        signature: felt*
+    )
+
+func execute(
+        to: felt,
+        selector: felt,
+        calldata_len: felt,
+        calldata: felt*,
+        nonce: felt
+    ) -> (response : felt):
+```
+
+> Notice we're omitting `initialize` from the API spec since we consider it a temporary artifact of the lack of `this.address` and we hope to remove it as soon as possible.
+
+#### `get_public_key`
+
+Returns the public key associated with the Account contract.
+
+##### Parameters:
+
+```
+None
+```
+
+##### Returns:
+```
+(implicit) syscall_ptr : felt*
+(implicit) pedersen_ptr: HashBuiltin*
+(implicit) range_check_ptr
+
+public_key: felt
+```
+
+#### `get_address`
+
+Returns the contract address of the Account.
+
+##### Parameters:
+
+    None
+
+##### Returns:
+
+```
+(implicit) syscall_ptr : felt*
+(implicit) pedersen_ptr: HashBuiltin*
+(implicit) range_check_ptr
+
+contract_address: felt
+```
+
+#### `get_nonce`
+
+Returns the current transaction nonce for the Account.
+
+##### Parameters:
+```
+None
+```
+##### Returns:
+
+```
+(implicit) syscall_ptr : felt*
+(implicit) pedersen_ptr: HashBuiltin*
+(implicit) range_check_ptr
+```
+
+#### `set_public_key`
+
+Sets the public key that will control this Account. It can be used to rotate keys for security, change them in case of compromised keys or even transferring ownership of the account.
+
+##### Parameters:
+```
+public_key: felt
+```
+
+##### Returns:
+```
+(implicit) syscall_ptr : felt*
+(implicit) pedersen_ptr: HashBuiltin*
+(implicit) range_check_ptr
+```
+
+#### `is_valid_signature`
+
+This function is inspired in [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) and checks whether a given signature is valid, otherwise it reverts.
```

Comment:
```suggestion
This function is inspired by [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) and checks whether a given signature is valid, otherwise it reverts.
```